src/main/java/org/sonarlint/intellij/ui/SonarLintRulePanel.java

```diff
@@ -140,4 +178,65 @@ public JComponent getPanel() {
   public void show() {
     panel.setVisible(true);
   }
+
+  public static class CustomHTMLEditorKit extends HTMLEditorKit {
+    private static HTMLFactory factory = null;
+
+    @Override
+    public ViewFactory getViewFactory() {
+      if (factory == null) {
+        factory = new HTMLFactory() {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Synchronize this lazy initialization of 'factory' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2444)